### DISCUSSION TITLE
Monitor the Pump Istwert of HKP-A01 and EQ-A02

### DISCUSSION THREAD

**Discussion by pbleij** [mbuchber/ha_heliotherm#37] [Q&A]
Is there possibility to monitor the pump Istwert of HKP-A01 and EQ-A02 of groundsource Heliotherm?
I would like to monitor if the maximum pump power is used during heating cycles but do not know how.
Could the integration with Heliotherm visualize this value?
![image](https://github.com/user-attachments/assets/1907238e-9a6c-4a01-ac0b-27713eacc7f0)

**Comment by mbuchber** [comment]
Hi, I have no idea, since I don't own such a pump... i just had one document (attached in the repo) about the Modbus-TCP registers, that I read and write according to the docs. So if you have infos about which registers, values and datatypes I could try - but cannot verify this in my home...
Best regards,
 Michael

**Reply by pbleij** [reply]
Thanks for you reaction. I will ask Heliotherm. No clue on outcome and I must admit that I know Modbus from only a 10.000ft helicopter view, not at all how to even program it to get that data.
Would you be aware that more data can be fetched in general beyond what is openly shared now as Modbus registers? Would every single setting you can find in the installer menu also be available via Modbus interface? will also ask heliotherm on that.

**Reply by pbleij** [reply]
Answer received. No Go for now. Only visible in Schreiber.

Sehr geehrter Herr Bleij

„Nein“ leider gibt es „kein“ Modbus – Register für die „Drehzahlen“ von „HKP-AO1“ & „EQ-AO2“.
Lediglich eine „Anzeige“, ob die „Pumpen / Ventilatoren“ laufen, also „0 & 1“.
Sie können dies aber auf „Remote Control“ beim „Schreiber“ überprüfen / auslesen.
Hier einfach „AO1“ & „AO2“ beim „Filter“, links oben, eingeben.

Freundliche Grüße / Kind Regards
|Technischer Support / Technical Support|
Heliotherm Wärmepumpentechnik Ges.m.b.H

**Reply by pbleij** [reply]
Next answer "if" this would be included into current "WebRegler-RCG" still was a No Go for now but they would forward to development.

Betreff: Re: Modbus Documentation request for new registers
Danke fur Deine Antwort.
Ist da eine R&D plannung um die werten im ModBus registers zu programmieren?

Mit freundliche grüße.
Pascal Bleij

Nein, für den WEB Regler + RCG „leider“ nicht.
Eventuell beim „neuen“ Regler der gerade „entwickelt“ wird.
Wir übergeben es aber trotzdem unserer Programmierabteilung, falls dies „doch noch“ irgendwann in Betracht gezogen wird.
Diese meldet sich bei Ihnen oder Ihrem Kompetenzpartner, sollte dem so sein.

Freundliche Grüße / Kind Regards
|Technischer Support / Technical Support|

**Reply by kobius77** [reply]
also wenn die Daten in der cloud protokolliert werden können dann müssen sie 1. aus dem WEB Regler raus und 2. über das RCG in die cloud kommen .. aber modbus-register gibts tatsächlich nicht dazu.

bei @dstrigl's htheatpump gibt es ein "EQP n_min(%)" .. https://github.com/dstrigl/htheatpump/blob/master/htheatpump/htparams-xxxxxx-3_0_20-273.csv#L478
das sollte im übrigen auch die Drehzahl vom Propeller bei einer Luft-Kiste liefern..

**Reply by pbleij** [reply]
I looked again in the table you referenced and see also the A01-HKP and A02-EQ registers.

A01-HKP:
https://github.com/dstrigl/htheatpump/blob/99b2c3192a4c6d318c897a64fb3de7eb83fb38a0/htheatpump/htparams-xxxxxx-3_0_20-273.csv#L49
And EQ-A02 below it.

Need to find out which files to update to get this registers 47 and 48 monitored in HA integration :-)

**Reply by kobius77** [reply]
this list does **not** contain modbus registers of the Heliotherm RCG (which is what officially is needed to talk "modbus" to a Heliotherm heatpump). 

dstrigl's software actually replaces the RCG.

**Comment by hstrohmaier** [comment]
I'll check this when I'm back home end of week. I'm running an HT Complete and it is easy for me to check the values of the registers. The latest docs I got from HT had no information about A01-HKP and EQ-A02.

**Comment by hstrohmaier** [comment]
I checked the document "20230705_Fachmannebene_RCGX_1.0.5.4_DE_mail.pdf" V1.6 from 24.04.2023 for "Remote Control Gateway X". In this doc registers 47 and 48 are used for this:

47 4 INT16 MKR1 Rücklauftemperatur Einheit in 0,1°C MP 64
48 4 INT16 MKR2 Vorlauftemperatur Einheit in 0,1°C MP 69_

If anyone has a newer doc for RCG 2.0.0.10 I'll update the registers.

**Comment by kobius77** [comment]
> I would like to monitor if the maximum pump power is used during heating cycles but do not know how.

and as i just happen to read your initial post:
if you're after a _power_ reading of the pump it might be easy(/ier) to just attach some kind of powermeter - easiest one with clamp(s) (like from [here](https://www.shelly.com/collections/energy-metering?sort_by=manual&sort_by=manual&filter.p.m.custom.product_type=Energy+Meters&filter.v.price.gte=&filter.v.price.lte=)) - with the added benefit that it would give real Amps/Watts readings which Heliotherm's built in Energymeter doesn't meter (assuming it's similar to my air source model, where the whole external part = EQ isn't metered)

**Comment by hstrohmaier** [comment]
Since release 2.1.0.5 the % value is the EQ pump is available on modbus. I have forked the project. Anyone who is interested is welcome to test it. I have made a contribution to @mbuchber and so this could also work directly via the ha_heliotherm integration at a later date.

**Reply by kobius77** [reply]
may i ask where you got that information from? is there a changelog available from HT? asked several times in the past - never got a reasonable reply.

**Reply by hstrohmaier** [reply]
I own a HT complete and I'm a developer familiar with many protocols. Modbus is no rocket science to me and if Icompare the registers and the values in remotecontrol.at I can guess ;)

**Reply by kobius77** [reply]
so the value came up on that remotecontrol thingy recently? never registered for that, only using the RCG locally/VPN.

**Reply by hstrohmaier** [reply]
You can get in touch with me by emailing hs@ategus.de – just make sure it's in German ;)

**Comment by hstrohmaier** [comment]
The value isn't new in the remote control, but it is new in the Modbus status register.
